GitCommands/Git/GitItemStatusFileExtensionComparer.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitCommands.Git
+{
+    /// <summary>
+    /// Compares the file extension of <see cref="GitItemStatus.Name"/> and then by <see cref="GitItemStatus.CompareTo(GitItemStatus)"/>.
+    /// </summary>
+    public class GitItemStatusFileExtensionComparer : Comparer<GitItemStatus>
+    {
+        public override int Compare(GitItemStatus x, GitItemStatus y)
+        {
+            var lhsExt = Path.GetExtension(x.Name);
```

Comment:
❗️  Input parameters must be validated, otherwise there is a risk of NRE

Reply:
Good catch.